lib/schemes/refresh.js

```diff
@@ -0,0 +1,375 @@
+    if (clientId) {
+      this._setClientId(clientId)
+    }
+
+    // End the refresh
+    this.isRefreshing = false
+  }
+
+  async _refreshToken () {
+    // Refresh endpoint is disabled
+    if (!this.options.endpoints.refresh) return
+
+    // Sync tokens
+    this._setToken(this.$auth.syncToken(this.name))
+    this.$auth.syncRefreshToken(this.name)
+
+    // Token is required but not available
+    if (!this.$auth.getToken(this.name)) return
+
+    // Token is already being refreshed
+    if (this.isRefreshing) return
+
+    // Start the refresh
+    this.isRefreshing = true
+
+    const { dataClientId, dataGrantType, grantType } = this.options
+    const endpoint = {
+      data: {
+        [this.options.dataRefreshToken]: this.$auth.getRefreshToken(this.name)
+      }
+    }
+
+    // Only add client id to payload if enabled
+    if (dataClientId) {
+      endpoint.data[dataClientId] = this._getClientId()
+    }
+
+    // Only add grant type to payload if enabled
+    if (dataGrantType !== false) {
+      endpoint.data[dataGrantType] = grantType
+    }
+
+    // Try to fetch user and then set
+    return this.$auth.requestWith(
+      this.name,
+      endpoint,
+      this.options.endpoints.refresh,
+      true
+    ).then(({ response, result }) => {
+      this._updateTokens(result)
+      return response
+    }).catch(error => {
+      this.$auth.callOnError(error, { method: 'refreshToken' })
+    })
+  }
+
+  _scheduleTokenRefresh () {
+    // If auto refresh is disabled, bail
+    if (!this.options.autoRefresh.enable) return
+
+    let intervalDuration = (this._getTokenExpiration() - Date.now()) * 0.75
+    if (intervalDuration < 1000) {
+      // in case you misconfigured refreshing this will save your auth-server from a self-induced DDoS-Attack
+      intervalDuration = 1000
+    }
+
+    this.refreshInterval = setInterval(() => {
+      this._refreshToken()
+    }, intervalDuration)
+  }
+
+  _initializeRequestInterceptor () {
+    this.$auth.ctx.app.$axios.onRequest(async config => {
+      // Sync tokens
+      const token = this.$auth.syncToken(this.name) || false
+      const refreshToken = this.$auth.syncRefreshToken(this.name)
+      this._setToken(token)
+
+      // If no token or no refresh token, bail
+      if (!token || !refreshToken) return config
+
+      // Update header
+      config.headers[this.options.tokenName] = token
+
+      const now = new Date().setMilliseconds(0)
+      const tokenExpiration = new Date(this._getTokenExpiration()).setMilliseconds(0)
+      let refreshTokenExpiration = this._getRefreshTokenExpiration()
+
+      if (refreshTokenExpiration) {
+        refreshTokenExpiration = new Date(refreshTokenExpiration).setMilliseconds(0)
+      }
+
+      if (!this.isRefreshing) {
+        if (now >= tokenExpiration) {
+          if (now < refreshTokenExpiration) {
+            // Refresh the token
+            return this._refreshToken().then(() => {
+              this._scheduleTokenRefresh()
```

Comment:
This can happen on server and after client will get response, server will try to update refresh token indefinitely. I suggest adding `if (!process.client) return` to _scheduleTokenRefresh, so it doesn't break anything if someone will make more _scheduleTokenRefresh calls on server.

Reply:
I tried this and the error still showing.
Any suggestions?
Thanks!

Reply:
This is currently being be refactored to use a sentral refresh controller. See https://github.com/nuxt-community/auth-module/pull/548

The problem above I think, is that `_refreshToken` is async, so the contents of `.then` are run after the request is sent, just as the error is saying. It could be solved with a simple `await`. But it does not matter, as it is to be refactored very soon.

Reply:
Fixed in commit: https://github.com/nuxt-community/auth-module/pull/361/commits/ac78c3171ee669fa0c8bbcd601c401bb90897647